src/libcore/ptr/mod.rs

```diff
@@ -1043,50 +1043,59 @@ pub unsafe fn write_volatile<T>(dst: *mut T, src: T) {
 /// Any questions go to @nagisa.
 #[lang = "align_offset"]
 pub(crate) unsafe fn align_offset<T: Sized>(p: *const T, a: usize) -> usize {
-    /// Calculate multiplicative modular inverse of `x` modulo `m`.
+    /// Calculate multiplicative modular inverse of `x` modulo `m = 2^mpow`.
     ///
     /// This implementation is tailored for align_offset and has following preconditions:
     ///
-    /// * `m` is a power-of-two;
+    /// * The requested modulu `m` is a power-of-two, so `mpow` can be an argument;
     /// * `x < m`; (if `x ≥ m`, pass in `x % m` instead)
     ///
+    /// It also leaves reducing the result modulu `m` to the caller, so the result may be larger
+    /// than `m`.
+    ///
     /// Implementation of this function shall not panic. Ever.
     #[inline]
-    fn mod_inv(x: usize, m: usize) -> usize {
+    fn mod_pow_2_inv(x: usize, mpow: usize) -> usize {
         /// Multiplicative modular inverse table modulo 2⁴ = 16.
         ///
         /// Note, that this table does not contain values where inverse does not exist (i.e., for
         /// `0⁻¹ mod 16`, `2⁻¹ mod 16`, etc.)
         const INV_TABLE_MOD_16: [u8; 8] = [1, 11, 13, 7, 9, 3, 5, 15];
         /// Modulo for which the `INV_TABLE_MOD_16` is intended.
         const INV_TABLE_MOD: usize = 16;
-        /// INV_TABLE_MOD²
-        const INV_TABLE_MOD_SQUARED: usize = INV_TABLE_MOD * INV_TABLE_MOD;
+        /// $s$ such that INV_TABLE_MOD = $2^s$.
```

Comment:
I prefer using normal backtick to document code. Latex seems to be nice and expressive but they don't look
well on plain code.
```suggestion
        /// `s` such that `INV_TABLE_MOD = 2^s`.
```